Options.py

```diff
@@ -797,17 +797,17 @@ class VerifyKeys(metaclass=FreezeValidKeys):
     verify_location_name: bool = False
     value: typing.Any
 
-    @classmethod
-    def verify_keys(cls, data: typing.Iterable[str]) -> None:
-        if cls.valid_keys:
-            data = set(data)
-            dataset = set(word.casefold() for word in data) if cls.valid_keys_casefold else set(data)
-            extra = dataset - cls._valid_keys
+    def verify_keys(self) -> None:
+        if self.valid_keys:
+            data = set(self.value)
+            dataset = set(word.casefold() for word in data) if self.valid_keys_casefold else set(data)
+            extra = dataset - self._valid_keys
             if extra:
-                raise Exception(f"Found unexpected key {', '.join(extra)} in {cls}. "
-                                f"Allowed keys: {cls._valid_keys}.")
+                raise Exception(f"Found unexpected key {', '.join(extra)} in {self}. "
```

Comment:
Using `self` instead of `cls` here might not be intended. `Option` doesn't define `__str__` (the only option in Options.py that does is `Range`), so falls back to `__repr__` defined on `Option`. I ?think? this is user-facing, and I don't know if we want the output of `__repr__` to be user-facing.

You could maintain the old behaviour with `self.__class__`. You could also define a `__str__` method on `Option`. When I tested, the repr seemed good enough, though I noted it used the option class name rather than the yaml option name or the display name:

```
Exception: Found unexpected key Wraithe in StartInventory(Additional Starting Vespene: 5, Wraithe: 1). Allowed Keys: <wall of text>.
```

Reply:
the old behavior is the exact same. options don't have access to the yaml/display name of the option.

Reply:
I must have gotten a bit mixed up as `current_option_name` accesses something, but checking again it can't access the option name as you say.

I'm happy to call this comment thread resolved.